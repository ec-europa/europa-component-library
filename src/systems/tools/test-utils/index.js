const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
