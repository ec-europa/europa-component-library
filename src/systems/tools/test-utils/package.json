{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
