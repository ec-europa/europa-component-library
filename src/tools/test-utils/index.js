const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
