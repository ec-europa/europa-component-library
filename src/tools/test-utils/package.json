{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
