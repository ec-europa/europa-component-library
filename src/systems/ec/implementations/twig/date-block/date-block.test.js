import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoData from './demo/data';

describe('EC - Date Block', () => {
  const template = '@ecl-twig/ec-component-date-block/ecl-date-block.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  test('renders correctly', () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra class names', () => {
    expect.assertions(1);

    const optionsWithExtraClasses = merge(demoData, {
      extra_classes: 'custom-class custom-class--test',
    });

    return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra attributes', () => {
    expect.assertions(1);

    const optionsWithExtraClasses = merge(demoData, {
      extra_attributes: [
        { name: 'data-test', value: 'data-test-value' },
        { name: 'data-test-1', value: 'data-test-value-1' },
      ],
    });

    return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
  });

  test('renders correctly with time set', () => {
    expect.assertions(1);

    const optionsWithTimeSet = merge(demoData, {
      date_time: '15:30',
    });

    return expect(render(optionsWithTimeSet)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra attributes and time set', () => {
    expect.assertions(1);

    const optionsWithExtraClassesAndTime = merge(demoData, {
      extra_attributes: [
        { name: 'data-test', value: 'data-test-value' },
        { name: 'data-test-1', value: 'data-test-value-1' },
      ],
      date_time: '15:30',
    });

    return expect(
      render(optionsWithExtraClassesAndTime)
    ).resolves.toMatchSnapshot();
  });

  test('with validation enabled and missing input data returns the right warning message', () => {
    expect.assertions(1);

    const dataCompliance = { ...demoData, _compliance_: true };
    dataCompliance.year = '';

    return expect(render(dataCompliance)).resolves.toMatchSnapshot();
  });
});
