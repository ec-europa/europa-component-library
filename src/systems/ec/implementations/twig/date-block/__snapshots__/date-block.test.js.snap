// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Date Block renders correctly 1`] = `
<test>
  <time
    class="ecl-date-block"
    datetime="2019-09-26"
  >
    <span
      class="ecl-date-block__day"
    >
      26
    </span>
    <abbr
      class="ecl-date-block__month"
      title="September"
    >
      Sep
    </abbr>
    <span
      class="ecl-date-block__year"
    >
      2019
    </span>
  </time>
</test>
`;

exports[`EC - Date Block renders correctly with extra attributes 1`] = `
<test>
  <time
    class="ecl-date-block"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
    datetime="2019-09-26"
  >
    <span
      class="ecl-date-block__day"
    >
      26
    </span>
    <abbr
      class="ecl-date-block__month"
      title="September"
    >
      Sep
    </abbr>
    <span
      class="ecl-date-block__year"
    >
      2019
    </span>
  </time>
</test>
`;

exports[`EC - Date Block renders correctly with extra attributes and time set 1`] = `
<test>
  <time
    class="ecl-date-block"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
    datetime="15:30"
  >
    <span
      class="ecl-date-block__day"
    >
      26
    </span>
    <abbr
      class="ecl-date-block__month"
      title="September"
    >
      Sep
    </abbr>
    <span
      class="ecl-date-block__year"
    >
      2019
    </span>
  </time>
</test>
`;

exports[`EC - Date Block renders correctly with extra class names 1`] = `
<test>
  <time
    class="ecl-date-block custom-class custom-class--test"
    datetime="2019-09-26"
  >
    <span
      class="ecl-date-block__day"
    >
      26
    </span>
    <abbr
      class="ecl-date-block__month"
      title="September"
    >
      Sep
    </abbr>
    <span
      class="ecl-date-block__year"
    >
      2019
    </span>
  </time>
</test>
`;

exports[`EC - Date Block renders correctly with time set 1`] = `
<test>
  <time
    class="ecl-date-block"
    datetime="15:30"
  >
    <span
      class="ecl-date-block__day"
    >
      26
    </span>
    <abbr
      class="ecl-date-block__month"
      title="September"
    >
      Sep
    </abbr>
    <span
      class="ecl-date-block__year"
    >
      2019
    </span>
  </time>
</test>
`;

exports[`EC - Date Block with validation enabled and missing input data returns the right warning message 1`] = `
<test>
  <!-- ecl2-compliance: date-block -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [date-block] - "year" is mandatory.
    </p>
  </div>
</test>
`;
